HabitTracker.harris-andy/Program.cs

```diff
@@ -0,0 +1,592 @@
+        if (count == 0)
+        {
+            PopulateDatabase();
+        }
+    }
+    connection.Close();
+}
+
+MainMenu();
+
+void MainMenu()
+{
+    Console.Clear();
+    bool closeApp = false;
+    bool withIds = false;
+    while (closeApp == false)
+    {
+        string message =
+            "--------------------------------------------------\n" +
+            "\n\t\tMAIN MENU\n\n" +
+            "\tWhat would you like to do?\n\n" +
+            "\tType 0 to Close Application\n" +
+            "\tType 1 to View All Records\n" +
+            "\tType 2 to Insert Record\n" +
+            "\tType 3 to Delete Record\n" +
+            "\tType 4 to Update Record\n" +
+            "\tType 5 to View A Record Summary\n" +
+            "\tType 6 to Delete All Records :(\n" +
+            "\tType 7 to Add 100 Rows of Fake Data\n" +
+            "--------------------------------------------------\n";
+
+        int inputNumber = -1;
+        while (inputNumber < 0)
+        {
+            inputNumber = validateNumberEntry(message, isMainMenu: true);
+            Console.Clear();
+        }
+
+        switch (inputNumber)
+        {
+            case 0:
+                Console.WriteLine("\nBye!\n");
+                closeApp = true;
+                Environment.Exit(0);
+                break;
+            case 1:
+                GetAllRecords(withIds);
+                break;
+            case 2:
+                Insert();
+                break;
+            case 3:
+                Delete();
+                break;
+            case 4:
+                Update();
+                break;
+            case 5:
+                GetRecordSummary();
+                break;
+            case 6:
+                DeleteTableContents();
+                break;
+            case 7:
+                PopulateDatabase();
+                break;
+            default:
+                Console.Clear();
+                Console.WriteLine("\nInvalid Command. Give me number!");
+                break;
+        }
+    }
+}
+
+void Insert()
+{
+    Console.Clear();
+    string date = GetDateInput();
+    Console.Clear();
+    string hobby = GetHobby();
+    Console.Clear();
+    string units = GetUnitsInput();
+    Console.Clear();
+    int quantity = GetQuantityInput();
+
+    using (var connection = new SqliteConnection(connectionString))
+    {
+        connection.Open();
+        using (var command = new SqliteCommand("INSERT INTO habits (Date, Hobby, Units, Quantity) VALUES (@date, @hobby, @units, @quantity)", connection))
+        {
+            command.Parameters.AddWithValue("@date", date);
+            command.Parameters.AddWithValue("@hobby", hobby);
+            command.Parameters.AddWithValue("@units", units);
+            command.Parameters.AddWithValue("@quantity", quantity);
+
+            command.ExecuteNonQuery();
+        }
+        connection.Close();
+    }
+}
+
+string GetDateInput()

```

Comment:
There is some craziness going on here. Whilst the solution works fine. Why not return a `DateTime` type? This might be useful in future cases where you might need to validate dates etc.
I only say this because if I see a method called `GetDateInput()` I would expect a date return type. Maybe `GetInputDateAsString()` would be a better name.